mongodbatlasv2/atlas_client.go

```diff
@@ -91,3 +92,22 @@ func UseUserAgent(userAgent string) ClientModifier {
 		return nil
 	}
 }
+
+// IsError checks if API returned known error type.
+func IsError(err error) (*Error, bool) {
+	var openapiError *GenericOpenAPIError
+	if ok := errors.As(err, &openapiError); !ok {
+		return nil, false
+	}
+	errModel := openapiError.Model()
+	return &errModel, true
+}
+
+// IsErrorCode returns true if the error contains the specific code.
+func IsErrorCode(err error, code string) bool {
+	mappedErr, _ := IsError(err)
+	if mappedErr == nil {
+		return false
+	}
+	return mappedErr.GetErrorCode() == code
+}
```

Comment:
should this be 
```suggestion
func (e GenericOpenAPIError) IsErrorCode(code string) bool {
	return e.GetError().GetErrorCode() == code
}
```

Reply:
I think that GenericOpenAPIError is an implementation detail and should never be part of the interface (used by the clients). Users interact with 2 methods that are fairly generic and return only what they want.